arbor/network_impl.cpp

```diff
@@ -0,0 +1,388 @@
+#include "network_impl.hpp"
+#include "cell_group_factory.hpp"
+#include "communication/distributed_for_each.hpp"
+#include "label_resolution.hpp"
+#include "network_impl.hpp"
+#include "threading/threading.hpp"
+#include "util/range.hpp"
+#include "util/spatial_tree.hpp"
+
+#include <Random123/threefry.h>
+
+#include <arbor/arbexcept.hpp>
+#include <arbor/benchmark_cell.hpp>
+#include <arbor/cable_cell.hpp>
+#include <arbor/common_types.hpp>
+#include <arbor/context.hpp>
+#include <arbor/domain_decomposition.hpp>
+#include <arbor/lif_cell.hpp>
+#include <arbor/load_balance.hpp>
+#include <arbor/morph/place_pwlin.hpp>
+#include <arbor/spike_source_cell.hpp>
+#include <arbor/util/unique_any.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <mutex>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <type_traits>
+#include <unordered_map>
+#include <vector>
+
+namespace arb {
+
+namespace {
+// We only need minimal hash collisions and good spread over the hash range, because this will be
+// used as input for random123, which then provides all desired hash properties.
+// std::hash is implementation dependent, so we define our own for reproducibility.
+
+std::uint64_t simple_string_hash(const std::string_view& s) {
```

Comment:
That's FNV-1a, right?

Reply:
Yes

Reply:
Maybe we add an `arb_internal_hash` (see #2215) and also tell which algo we're using.

Reply:
Ok, this is convincing ;) If it's used for 'seeding' the cbRNG, wouldn't we want something stronger?